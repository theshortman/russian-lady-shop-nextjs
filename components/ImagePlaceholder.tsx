import { FC } from "react";

interface ImagePlaceholderProps {
  text: string;
}

const ImagePlaceholder: FC<ImagePlaceholderProps> = ({ text }) => {
  return (
    <svg
      className="img-fluid img-thumbnail"
      width="600"
      height="912"
      xmlns="http://www.w3.org/2000/svg"
      role="img"
      aria-label="No photo"
      preserveAspectRatio="xMidYMid slice"
      focusable="false"
    >
      <rect width="100%" height="100%" fill="#ffffff"></rect>
      <text x="50%" y="50%" fill="#dee2e6" dy=".3em" textAnchor="middle">
        {text}
      </text>
    </svg>
  );
};

export default ImagePlaceholder;
