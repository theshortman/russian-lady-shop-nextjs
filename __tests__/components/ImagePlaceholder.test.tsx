import { render } from "@testing-library/react";
import ImagePlaceholder from "@/components/ImagePlaceholder";

test("renders ImagePlaceholder unchanged", () => {
  const { container } = render(<ImagePlaceholder text="Test" />);
  expect(container).toMatchSnapshot();
});
