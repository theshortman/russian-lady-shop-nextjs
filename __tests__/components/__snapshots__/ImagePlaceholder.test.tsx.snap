// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ImagePlaceholder unchanged 1`] = `
<div>
  <svg
    aria-label="No photo"
    class="img-fluid img-thumbnail"
    focusable="false"
    height="912"
    preserveAspectRatio="xMidYMid slice"
    role="img"
    width="600"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="#ffffff"
      height="100%"
      width="100%"
    />
    <text
      dy=".3em"
      fill="#dee2e6"
      text-anchor="middle"
      x="50%"
      y="50%"
    >
      Test
    </text>
  </svg>
</div>
`;
